src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -267,8 +267,7 @@ void MetaspaceShared::initialize_for_static_dump() {
   size_t symbol_rs_size = LP64_ONLY(3 * G) NOT_LP64(128 * M);
   _symbol_rs = ReservedSpace(symbol_rs_size);
   if (!_symbol_rs.is_reserved()) {
-    vm_exit_during_initialization("Unable to reserve memory for symbols",
-                                  err_msg(SIZE_FORMAT " bytes.", symbol_rs_size));
+    MetaspaceShared::unrecoverable_writing_error(err_msg("Unable to reserve memory for symbols\n%ld bytes.", symbol_rs_size));
```

Comment:
For complex message, it would be better to do this:

```
   log_error(cds)(Unable to reserve memory for symbols\n%ld bytes.", symbol_rs_siz);
   MetaspaceShared::unrecoverable_writing_error();
```

The `message` argument for `unrecoverable_writing_error()` should be used only for simple, fixed messages.